### ISSUE TITLE
Consider where to perform referential integrity checks

### ISSUE THREAD

**Issue by jeffbaumes** [microbiomedata/nmdc-metadata#309]
Not sure where this code should lie but it should happen before ingestion into the pilot system.

**Comment by jeffbaumes** [comment]
@wdduncan Also assigning you and @jbeezley to this one. Similar question, but I don't believe we can do this check in the schema. Where should we put non-schema validation code?

**Comment by wdduncan** [comment]
Thanks @jeffbaumes !

Adding @dehays, @cmungall , and @dwinston to this tickets. This gets into larger issues ...

**Comment by ssarrafan** [comment]
> Thanks @jeffbaumes !
> 
> Adding @dehays, @cmungall , and @dwinston to this tickets. This gets into larger issues ...

@wdduncan Can the "larger issues" be broken down into smaller chunks?  Let me know if I should create additional GitHub issues related to this.

**Comment by dehays** [comment]
I agree with two things @jeffbaumes is saying here:  1) Ingest into the search application should not have responsibility for dealing with referential integrity as it does now.  This should happen upstream. 2) I don't think this can be done in the schema.  The schema can require that the was_informed, has_input, had_output have values but I don't think the capability is there to guarantee the existence of the referenced entities.

That leaves this type of business logic to be applied in or around the metadata store.  If that is Mongo that means audit logic outside Mongo that checks for existence of referenced entities across the the Mongo document collections.  If that is TerminusDB then if I am understanding @dwinston correctly it is possible to apply referential integrity checking rules.

**Comment by ssarrafan** [comment]
Moved to May sprint per Bill and added @dwinston as an assignee.

**Comment by ssarrafan** [comment]
Based on the meeting today with @dehays, @emileyfadrosh, @dwinston, @wdduncan, and @jbeezley I'm closing this issue.  We had the discussion to consider where to do referential integrity checks. The plan is to update the NMDC schema by adding required fields needed (will create a ticket for @wdduncan) and to apply a rule that uses the updated schema changes and applies I to the database (will create a ticket for @wdduncan and @dwinston) .  

Note: Consider continuing to group schema changes into a version and update the changelog and version appropriately. May want to consider only doing this quarterly? 

Note: This work requires writing logic in Python. Additional help would be nice to get and a new issue will be created for @emileyfadrosh and Set to get some help for Bill and Donny.  

Note: A new ticket will be created for @dwinston who said he will document and template this type of work to make it easier for others to help with these types of changes in the future.
